main.tf

```diff
@@ -0,0 +1,314 @@
+resource "null_resource" "enable_destination_api" {
+  provisioner "local-exec" {
+    command = "sh ${path.module}/scripts/enable-api.sh ${local.destination_api} ${local.destination_project}"
+  }
+}
+
+# Pub/Sub topic
+resource "google_pubsub_topic" "topic" {
+  count   = "${local.destination_type == "pubsub" ? 1 : 0}"
+  name    = "${local.destination_name}"
+  project = "${local.destination_project}"
+
+  depends_on = [
+    "null_resource.enable_destination_api",
+  ]
+}
+
+# BigQuery dataset
+resource "google_bigquery_dataset" "dataset" {
+  count      = "${local.destination_type == "bigquery" ? 1 : 0}"
+  dataset_id = "${local.destination_name}"
+  project    = "${local.destination_project}"
+
+  # Delete all tables in dataset on destroy.
+  # This is required because a dataset cannot be deleted if it contains data
+  provisioner "local-exec" {
+    when    = "destroy"
+    command = "sh ${path.module}/scripts/delete-bq-tables.sh ${local.destination_project} ${local.destination_name}"
+  }
+
+  depends_on = [
+    "null_resource.enable_destination_api",
+  ]
+}
+
+# Storage bucket
+resource "google_storage_bucket" "bucket" {
+  count         = "${local.destination_type == "storage" ? 1 : 0}"
+  name          = "${local.destination_name}"
+  project       = "${local.destination_project}"
+  storage_class = "MULTI_REGIONAL"
+  location      = "US"
+  force_destroy = true
+
+  depends_on = [
+    "null_resource.enable_destination_api",
+  ]
+}
+
+#---------#
+# Logsink #
+#---------#
+
+# Project-level
+resource "google_logging_project_sink" "sink" {
+  count                  = "${local.is_project_level ? 1 : 0}"
+  name                   = "${var.name}"
+  project                = "${var.project}"
+  filter                 = "${var.filter}"
+  destination            = "${local.destination_uri}"
+  unique_writer_identity = "${var.unique_writer_identity}"
+
+  depends_on = [
+    "google_storage_bucket.bucket",
+    "google_pubsub_topic.topic",
+    "google_bigquery_dataset.dataset",
+  ]
+}
+
+# Folder-level
+resource "google_logging_folder_sink" "sink" {
+  count            = "${local.is_folder_level ? 1 : 0}"
+  name             = "${var.name}"
+  folder           = "${var.folder}"
+  filter           = "${var.filter}"
+  include_children = "${var.include_children}"
+  destination      = "${local.destination_uri}"
+
+  depends_on = [
+    "google_storage_bucket.bucket",
+    "google_pubsub_topic.topic",
+    "google_bigquery_dataset.dataset",
+  ]
+}
+
+# Org-level
+resource "google_logging_organization_sink" "sink" {
+  count            = "${local.is_org_level ? 1 : 0}"
+  name             = "${var.name}"
+  org_id           = "${var.org_id}"
+  filter           = "${var.filter}"
+  include_children = "${var.include_children}"
+  destination      = "${local.destination_uri}"
+
+  depends_on = [
+    "google_storage_bucket.bucket",
+    "google_pubsub_topic.topic",
+    "google_bigquery_dataset.dataset",
+  ]
+}
+
+# Billing-level
```

Comment:
Is billing level not working for now? I'm fine with moving it to a later PR but we shouldn't publish commented-out code.